docs/userguide/apps.rst

```diff
@@ -3,107 +3,100 @@
 Apps
 ====
 
-An **app** is a Parsl construct for representing a fragment of Python code 
-or external Bash shell code that can be asynchronously executed.
+An **App** defines a computation that will be executed asynchronously by Parsl.
+Apps are Python functions marked with a decorator which
+designates that the function will run asynchronously and cause it return
+a :class:`~concurrent.futures.Future` instead of the result.
 
-A Parsl app is defined by annotating a Python function with a decorator: 
-the ``@python_app`` decorator for a **Python app**, the ``@bash_app`` decorator for a **Bash app**,
-and the ``@join_app`` decorator for a **Join app**.
+Apps can be one of three types of functions, each with their own type of decorator
 
-Python apps encapsulate pure Python code, while Bash apps wrap calls to external applications and scripts,
-and Join apps allow composition of other apps to form sub-workflows.
+- ``@python_app``: Most Python functions
+- ``@bash_app``: A python function which returns a command line program to execute
+- ``@join_app``: A function which returns launches other apps
 
-Python and Bash apps are documented below. Join apps are documented in a later
-section (see :ref:`label-joinapp`)
+The intricacies of Python and Bash apps are documented below. Join apps are documented in a later
+section (see :ref:`label-joinapp`).
 
 Python Apps
 -----------
 
-The following code snippet shows a Python function ``double(x: int)``, which returns double the input
-value. 
-The ``@python_app`` decorator defines the function as a Parsl Python app.  
-
 .. code-block:: python
 
-       @python_app
-       def double(x):
-           return x * 2
+    @python_app
+    def hello_world(name: str) -> str:
+        return f'Hello, {name}!'
 
-       double(42)
+    print(hello_world('user').result())
 
-As a Parsl Python app is executed asynchronously, and potentially remotely, the function
-cannot assume access to shared program state. For example, it must explicitly import any 
-required modules and cannot refer to variables used outside the function. 
-Thus while the following code fragment is valid Python, it is not valid Parsl, 
-as the ``bad_double()`` function requires the `random` module and refers to the external 
-variable ``factor``.
 
-.. code-block:: python
+Python Apps run Python functions. The code inside a function marked by ``@python_app`` will run on a remote system.
```

Comment:
"will run on a remote system" --- of course, in this example things are local. Might be better to say something like "is what will be executed either locally or on the remote system"

Reply:
Great point.